src/lib/tauri/http/Response.ts

```diff
@@ -0,0 +1,51 @@
+import * as http from '@tauri-apps/api/http';
+import { either, flow, iots } from '@code-expert/prelude';
+import { HttpError, httpError } from './HttpError';
+
+const ResponseError = iots.strict({
+  statusCode: iots.number,
+  error: iots.string,
+  message: iots.string,
+});
+type ResponseError = iots.TypeOf<typeof ResponseError>;
+
+const Response = iots.union([
+  iots.strict({
+    ok: iots.literal(true),
+    data: iots.unknown,
+  }),
+  iots.strict({
+    ok: iots.literal(false),
+    data: ResponseError,
+  }),
+]);
+
+export const parseResponse: <A>(
+  codec: iots.Decoder<unknown, A>,
+) => (r: http.Response<unknown>) => either.Either<HttpError, A> = (codec) =>
+  flow(
+    Response.decode,
+    either.mapLeft(iots.formatValidationErrors),
+    either.getOrThrow(
+      (errs) =>
+        new Error(`Fetch response did not match Tauri’s "Response" type': ${errs.join('; ')}`),
+    ),
+    (res) => (res.ok ? either.right(res.data) : either.left(toHttpError(res.data))),
+    either.map(
+      flow(
+        codec.decode,
+        either.mapLeft(iots.formatValidationErrors),
+        either.getOrThrow(
+          (errs) => new Error(`Response payload did not match codec: ${errs.join('; ')}`),
```

Comment:
this looks very convenient, but in the spirit of separating Tauri from 'our' http stuff, shouldn't we do the `throw` in our code?

Reply:
to keep lib/tauri pure and enact our error handling (throw + either where necessary) in our own code, I mean

Reply:
but it's probably easier to clean up these errors in your Exception-Handling branch